versions/3.1.1.md

```diff
@@ -2698,14 +2693,23 @@ components:
 
 #### <a name="discriminatorObject"></a>Discriminator Object
 
-When request bodies or response payloads may be one of a number of different schemas, a `discriminator` object can be used to aid in serialization, deserialization, and validation.  The discriminator is a specific object in a schema which is used to inform the consumer of the document of an alternative schema based on the value associated with it.
+When request bodies or response payloads may be one of a number of different schemas, a `discriminator` object can be used to aid in serialization, deserialization, and validation. The `discriminator` keyword is used to inform the consumer of the document which of the alternatives is expected or preferred.
 
-When using the discriminator, _inline_ schemas will not be considered.
+`discriminator` uses a schema's "name" to automatically map a property value to
+a schema. The schema's "name" is the property name used when declaring the
+schema as a component in an OpenAPI document. For example, the name of the
+schema at `#/components/schemas/Cat` is "Cat". Therefore, when using
+`discriminator`, _inline_ schemas will not be considered because they don't have
+a "name".
+
+Schema names are scoped to the OpenAPI document they are defined in. That means
+if OpenAPI document "A" references a schema in OpenAPI document "B", the schema
+names in "A" aren't considered when evaluating the referenced schema in "B".
```

Comment:
This is the main part that's controversial.  In Swagger 2.0, the references were strongly implied to be scoped to document "A", and many tools implement that behavior.  There are other component name connections that have similar ambiguities and we should address them all at once.  (FWIW, I strongly agree with what you've written here, but it would be a de-facto breaking change for some and we recently decided not to do "clarifications" that are de-facto breaking changes).